compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -80,12 +80,52 @@ macro_rules! define_dep_nodes {
         }
 
         /// This enum serves as an index into arrays built by `make_dep_kind_array`.
-        #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, Encodable, Decodable)]
+        #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
```

Comment:
Please add a short comment here explaining why `u16` is used, e.g. some of the text from the PR's description will probably suffice.